services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/_core/_events_user_services.py

```diff
@@ -105,10 +103,26 @@ async def create_user_services(  # pylint: disable=too-many-statements
     )
 
     _logger.debug(
-        "Starting containers %s with compose-specs:\n%s",
+        "Submitting to %s it's compose-specs:\n%s",
         scheduler_data.service_name,
         compose_spec,
     )
+    await sidecars_client.submit_docker_compose_spec(
+        dynamic_sidecar_endpoint, compose_spec=compose_spec
+    )
+    scheduler_data.dynamic_sidecar.was_compose_spec_submitted = True
```

Comment:
Q: if `scheduler_data.dynamic_sidecar.was_compose_spec_submitted==True` would you execute `await sidecars_client.submit_docker_compose_spec` ? or asked differently, should you also check before submitting?

Reply:
The `SendUserServicesSpec` already checks that the spec was not submitted. Have a look at it's `will_trigger` implementation.